### ISSUE TITLE
Rule import/no-deprecated wot working

### ISSUE THREAD

**Issue by PainKKKiller** [import-js/eslint-plugin-import#2245]
Sorry maybe I missed something in docs but I am unable to make work rule import/no-deprecated. 
I installed eslint-plugin-import version 6.11.3
I put rules in my config
.
`
...
 import/no-deprecated: 1,
 import/no-cycle: 1
...
`
The problem is - I see rule no-cycle is working, and I can change level of it from warning to error. But I see no signs of import/no-deprecated rule despite the fact that entities marked with @deprecated tag in JSDocs exist in project. What can be wrong? Thanks in advance.

**Comment by PainKKKiller** [comment]
Sorry I did mistake I use eslint-plugin-import@2.21.0.
And one more details, I use it with typescript.
And code. For example we are using deprecated entity Accordeon.
Place where it used
`
import { Accordeon } from '@tfido-ui/business/Layout';
`
File in /node_modules/@tfido-ui/business/Layout.js:
`
export * from './Accordion';

`
File Accordion.js:
`
import * as React from 'react';

/**
* Аккордеон
*
* @deprecated
*
`* @params {any}
*/
const Accordeon: React.FC<any> = props => {
  // code
}

Accordeoin.displayName = 'Accordeon';

export { Accordeon };
`

**Comment by connorjs** [comment]
This appears to be a TypeScript-specific issue.

If I mark a const as `@deprecated` with JSDoc*, then this rule

1. Does emit an error/warn IF the source file is `.js`
2. Does NOT emit an error/warn IF the source file is `.ts`

I assume this relates to how TypeScript ESLint plugin parses JSDoc/TSDoc.

\*JSDoc - which is also TSDoc here?

_Just leaving some notes in case someone can correct me here. For now, I will look further to confirm this._

**Comment by ljharb** [comment]
@connorjs it's certainly possible that the TS eslint parser uses a different node name in .ts than in .js - that would be trivial to fix once it could be reproduced.

**Comment by connorjs** [comment]
I took a glance to see effort of _potentially_ implementing a fix (if TypeScript was the root cause). Looks like ESLint deprecated JSDoc parsing a while ago: <https://eslint.org/blog/2018/11/jsdoc-end-of-life/>.

This package uses [doctrine](https://github.com/eslint/doctrine) which is the deprecated/EOL-ed package from ESLint.

**Comment by connorjs** [comment]
Thanks @ljharb! I will look at how to reproduce in this project’s unit tests. Looks like there’s enough test structure to simulate this. Roughly,

1. Create TS file for no-deprecated
2. Add test cases that import that TS file

**Comment by connorjs** [comment]
_Off-topic: Relates to contributing._

@ljharb - What node version should I use for this package? I’m getting the following error when running. Looking at `travis.yml`, I had guessed node14 would work. I didn’t see anything in the CONTRIBUTING, but whoops if I missed another spot.

```
/Users/connorjs/w/connorjs/eslint-plugin-import/tests/src/cli.js:4
import path from 'path';
^^^^^^

SyntaxError: Cannot use import statement outside a module
```

**Comment by ljharb** [comment]
@connorjs anything that’s in the package.json engines.node. However, you’ll need to use the script commands in package.json to run tests properly.
